carbon-design-system/carbon
# Fix (16747)(A11y) : Proper Conveyance of Word/Character Count by screen readers

Closes #16747

When the `maxCount` word limit is set along with the `enableCounter` prop, the screen reader announcement for the word/character count used to only triggers once when `10 words/characters` were left. 

Also, no announcements are made when the `maximum` word/character count is reached. This inconsistency makes it difficult for screen reader users to understand the remaining count and when the limit is hit.

**This PR does two things** 

1. Incremental Announcements 
It will now make announcements for each of the last 10 characters/words, properly conveying the remaining count.

2. Max Count Announcement
An additional announcement will be made when the maximum word/character count is reached, ensuring users are informed when the limit is hit.

#### Changelog

The `useAnnouncer` function is updated to track when the maximum limit is reached and ensure announcements are made for each of the last 10 characters/words.

The test cases for `useAnnouncer` have been updated to reflect the new changes and ensure proper functionality.

Within the component, useAnnouncer is called inside a `useEffect` hook to manage the timing of announcements.
This is to ensures that same announcements are not repeated unnecessarily and are made after a delay, allowing both the counter and the announcement of the actual text being written to be audible



#### Testing / Reviewing

1. Vist textArea  playground in deployPreview
2. Enable enableCounter prop
3. Set counterMode to character/word (test for both)
4. Set maxCount to 12 
5. Type character/word: 
6. Run JAWS/VoiceOver and verify that the announcements are made for each of the last 10 characters/words and also verify that Max Count reached announcement is made

Also verify that readOnly status is announced by screenReader when readonly is true
